src/lib/apexPmd.ts

```diff
@@ -128,24 +128,30 @@ export class ApexPmd {
     const rulesetsArg = this.rulesets.join(',');
 
     const cacheKey = enableCache ? `--cache "${cachePath}"` : '--no-cache';
+    const noProgressBar = '--no-progress';
     const formatKey = `-f csv`;
     const targetPathKey = `-d "${targetPath}"`;
     const rulesetsKey = `-R "${rulesetsArg}"`;
 
-    const pmdKeys = `${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
+    const pmdKeys = `${noProgressBar} ${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
 
     const classPath = [
-      path.join(pmdBinPath, 'lib', '*'),
       path.join(workspaceRootPath, '*'),
       ...additionalClassPaths,
     ].join(CLASSPATH_DELM);
 
-    const javaExc = this.config.jrePath ? path.join(this.config.jrePath, 'bin', 'java') : 'java';
-    const cmd = `${javaExc} -cp "${classPath}" net.sourceforge.pmd.PMD ${pmdKeys}`;
+    let env : NodeJS.ProcessEnv = {};
+    env["CLASSPATH"] = classPath;
+    if (this.config.jrePath) {
+      env["JAVA_HOME"] = this.config.jrePath;
+    }
+
+    const cmd = `${pmdBinPath}/bin/pmd check ${pmdKeys}`;
```

Comment:
Calling the CLI script here. This works for sure under linux/macos. Under Windows, I don't know - needs to be tested. In theory, Windows should then execute pmd.bat...

Reply:
I've meanwhile checked this under Windows (after waiting a very long time until Windows finished the updates): But it is working. So, we don't need a platform specific command line here.